functorch/_src/aot_autograd.py

```diff
@@ -1488,18 +1488,30 @@ def create_aot_dispatcher_function(
     if config.use_dynamic_shapes:
         assert config.use_fake_tensor, "Dynamic shapes only works with fake tensor"
 
-    shape_env = ShapeEnv() if config.use_dynamic_shapes else None
-    fake_mode = FakeTensorMode(shape_env=shape_env) if config.use_fake_tensor else nullcontext()
+    # Check flat_args to see if they're already fake.  If so, use that fake
+    # mode instead.
+
+    for x in flat_args:
+        if isinstance(x, FakeTensor):
+            fake_mode = x.fake_mode
+            break
```

Comment:
I have a util I wrote for this, in the parent PR

```
def fake_mode_from_tensors(inputs: List[Any]):
    """
    Takes a list of anything, unflattened is fine, returns a fake_mode
    if any are fake. All fake modes on all fake tensors must be identical.
    Returns None if no fake_mode is fine
    """
    flat_inputs, _ = tree_flatten(inputs)
    fake_mode = None
    for flat_input in flat_inputs:
        if isinstance(flat_input, torch._subclasses.FakeTensor):
            if fake_mode is None:
                fake_mode = flat_input.fake_mode
            else:
                assert fake_mode == flat_input.fake_mode
    return fake_mode
```

Feel free to replace the continue-and-check-all-same-fake-mode assert w/ a break as you have it, but I do thing this should be in utils

Reply:
I didn't use this utility because `flat_args` is guaranteed to be a flat list of tensors here, whereas the utility does a flatten first. Better not to use pytree if you don't need it.